### ISSUE TITLE
Setting up Longpoll without Blocking

### ISSUE THREAD

**Issue by bdearborn** [l2ig/aqua#46]
Hi,

I am trying to setup a longpoll example, but I am having an issue with subsequent requests being blocked. Can you take a look at the following example?

**Test Endpoints**
-start the longpoll request (waits up to 10 seconds)
http://localhost/longpoll/

-test (doesn't return until longpoll is done)
http://localhost/test/

-toggle longpoll variable
http://localhost/toggle/

**Example Code**
```
//imports
import Aqua from "https://deno.land/x/aqua/mod.ts";

//global variables
const app = new Aqua(80);
let bChange = false;

//test
app.get("/test/", async (req) => {
    
    bChange = (bChange ? false : true);

    return {
        statusCode: 200,
        headers: { "Content-Type": "application/json" },
        content: JSON.stringify('test')
    };
});

//toggle
app.get("/toggle/", async (req) => {
    
    bChange = (bChange ? false : true);

    return {
        statusCode: 200,
        headers: { "Content-Type": "application/json" },
        content: JSON.stringify(bChange)
    };
});

//longpoll
app.get("/longpoll/", async (req) => {

    await waitFor(()=>bChange, 10000, 100);

    return {
        statusCode: 200,
        headers: { "Content-Type": "application/json" },
        content: JSON.stringify("Done polling")
    };
});

//functions
function sleep(ms) { return new Promise(resolve => setTimeout(resolve, ms));    }
async function waitFor(test, timeout_ms = 20*2000, frequency = 100){
    return new Promise( async (resolve,reject) => {
        if( typeof(timeout_ms) != "number") reject("Timeout argument not a number in waitFor(test, [timeout_ms], [frequency])");
        let result
        // wait until the result is truthy, or timeout
        while( result === undefined || result === false || result === null || result.length === 0 ){  // for non arrays, length is undefined, so != 0
            timeout_ms -= frequency;
            if( timeout_ms < 0 ){
                resolve(false);
                return;
            }
            await sleep(frequency);
            result = test();
        }
        // return result if test passed
        resolve(result);
    });
}
```

**Comment by l2ig** [comment]
Could you explain a bit more what's the exact issue? (What should happen and what does)

**Comment by l2ig** [comment]
Ahh got it, will investigate. Thanks so much for opening this issue! :+1:

**Comment by l2ig** [comment]
Thanks again for opening this issue. This has been fixed in v1.0.22 (5a95e0c41cf6e36bb4aba73c6d94505fc10d4418).
Super glad you discovered this :100:

**Comment by bdearborn** [comment]
Thanks!
